R/bench-summ.R

```diff
@@ -0,0 +1,166 @@
+#' Function that provides summary and plots for the benchmark run
+#'
+#' @param .res_tibble formatted bench result tibble with roc and prc columns
+#' roc column: Reciever Operator Curve Results (calculated with yardstick)
+#' prc column: Precision-Recall Curve Results (calculated with yardstick)
+#' @return A summary list with TF coverage, ROC, AUROC, PRAUC, Run time,
+#' ROC plots, and Heatmap plots
+#' @import ggplot2 pheatmap
+#' @importFrom rlang .data
+get_bench_summary <- function(.res_tibble) {
+  # get roc results
+  roc <- format_roc(.res_tibble, "roc")
+
+  # get PR roc results
+  pr <- format_roc(.res_tibble, "prc")
+
+  # Plot ROC
+  roc_plot <- ggplot(roc, aes(x = 1-specificity,
+                              y = sensitivity,
+                              colour = run_key)) +
+    geom_line() +
+    geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
+    xlab("FPR (1-specificity)") +
+    ylab("TPR (sensitivity)")
+
+  # Plot PR ROC
+  pr_plot <- ggplot(pr, aes(x = recall, y = precision , colour = run_key)) +
+    geom_line() +
+    geom_abline(intercept = 0, slope = 1, linetype = "dashed") +
```

Comment:
for PR plots a diagonal line is meaningless, we need a horizontal line with crosses the y-axis at the proportion of true positives (random performance)